### PR TITLE
Support Brave wallet

### PR DESCRIPTION
**Pull request by hungdoansy**

### CODE REVIEW

File: src/components/WalletModal/index.tsx [hunk 1]
```diff
@@ -231,102 +240,106 @@ export default function WalletModal({
...
-            active={option.connector === connector}
-            color={option.color}
-            link={option.href}
-            header={option.name}
-            subheader={null} //use option.descriptio to bring back multi-line
-            icon={require(`../../assets/images/${isDarkMode ? '' : 'light-'}${option.iconName}`).default}
-          />
         )
-      )
-    })
```
Comment (hungdoansy) [inline src/components/WalletModal/index.tsx:329]: Nothing was changed, I added `filter(Boolean)` only

File: src/utils/checkForBraveBrowser.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// https://stackoverflow.com/questions/36523448/how-do-i-tell-if-a-user-is-using-brave-as-their-browser
+let isBraveBrowser = false
+;(() => {
```
Comment (nguyenhoaidanh) [inline src/utils/checkForBraveBrowser.ts:3]: this is shorter.
```
;(async () => {
  isBraveBrowser = (navigator.brave && (await navigator.brave.isBrave())) || false
})()
```

Comment (hungdoansy) [reply src/utils/checkForBraveBrowser.ts:3]: 🙌

Comment (nguyenhoaidanh) [reply src/utils/checkForBraveBrowser.ts:3]: I mean you can write that code directly in IIFE, no need to create function `check` . so there is only 1 line in IIFE function

Comment (hungdoansy) [reply src/utils/checkForBraveBrowser.ts]: Yeah, agree

File: src/components/WalletModal/PendingView.tsx [hunk 3]
```diff
@@ -67,41 +64,31 @@ const LoadingWrapper = styled.div`
...
+  onClickTryAgain,
 }: {
-  connector?: AbstractConnector
-  error?: boolean
-  setPendingError: (error: boolean) => void
-  tryActivation: (connector: AbstractConnector) => void
+  walletOptionKey?: string // key of SUPPORTED_WALLETS
+  hasError?: boolean
+  renderHelperText?: () => React.ReactNode
+  onClickTryAgain: () => any
```
Comment (nguyenhoaidanh) [inline src/components/WalletModal/PendingView.tsx:75]: I don't see that func return anything. so `()=> void`

**Comment by nguyenhoaidanh** [comment]
LGTM, but 
Has the QC team tested done?

**Comment by hungdoansy** [comment]
> LGTM, but Has the QC team tested done?

Not yet, review needs to be done first

**Comment by nguyenhoaidanh** [comment]
> Not yet, review needs to be done first
u can send QC test now. after fix bug we can review again for you :v

File: src/components/WalletModal/PendingView.tsx (hunk 3 above)
Comment (hungdoansy) [reply src/components/WalletModal/PendingView.tsx]: Updated

**Comment by nguyenhoaidanh** [comment]
QC tested done, right?

File: src/components/AccountDetails/index.tsx [hunk 4]
```diff
@@ -223,33 +231,43 @@ export default function AccountDetails({
...
-        <IconWrapper size={20}>
-          <img src={FortmaticIcon} alt={'fortmatic logo'} />
-        </IconWrapper>
-      )
-    } else if (connector === portis) {
-      return (
-        <>
+    switch (connector) {
+      case injected:
+      case coin98InjectedConnector:
```
Comment (nguyenhoaidanh) [inline src/components/AccountDetails/index.tsx:236]: nice

**Comment by hungdoansy** [comment]
> QC tested done, right?

Still has some bugs =))

**Review by nguyenhoaidanh — APPROVED** [review]
lgtm

**Review by viet-nv — APPROVED** [review]

**Pull request merged**